js/ui/grid_core/ui.grid_core.keyboard_navigation.js

```diff
@@ -461,15 +478,38 @@ const KeyboardNavigationController = core.ViewController.inherit({
         }
 
         if(isOriginalHandlerRequired) {
-            this.getController('editorFactory').loseFocus();
+            this._editorFactory.loseFocus();
             if(this._editingController.isEditing() && !this._isRowEditMode()) {
-                this._resetFocusedCell();
+                this._resetFocusedView();
                 this._closeEditCell();
             }
         } else {
             eventArgs.originalEvent.preventDefault();
         }
     },
+    _processVirtualPosition: function(direction) {
+        const scrollable = this.component.getScrollable();
+        const columnIndex = this.getColumnIndex();
+
+        if(direction === 'next') {
+            const columnsCount = this._getVisibleColumnCount();
+            const nextColumnIndex = columnIndex + 1;
+
+            if(columnsCount === nextColumnIndex) {
+                const scrollOffset = scrollable && scrollable.scrollOffset();
+                const scrollOffsetLeft = scrollOffset && scrollOffset.left;
+                scrollable.scrollTo({ x: -scrollOffsetLeft });
```

Comment:
Why is not { x: 0 }?

Reply:
What is about rtl?

Reply:
>Why is not { x: 0 }?

0 should be there.

>What is about rtl?

RTL operates properly.